File: authentication/dotnet/azure/src/AzureIdentityAuthenticationProvider.cs
// ------------------------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
// ------------------------------------------------------------------------------

using Azure.Core;
using Microsoft.Kiota.Abstractions.Authentication;

namespace Microsoft.Kiota.Authentication.Azure;
/// <summary>
/// The <see cref="BaseBearerTokenAuthenticationProvider"/> implementation that supports implementations of <see cref="TokenCredential"/> from Azure.Identity.
/// </summary>
public class AzureIdentityAuthenticationProvider : BaseBearerTokenAuthenticationProvider
{
    /// <summary>
    /// The <see cref="AzureIdentityAuthenticationProvider"/> constructor
    /// </summary>
    /// <param name="credential">The credential implementation to use to obtain the access token.</param>
    /// <param name="allowedHosts">The list of allowed hosts for which to request access tokens.</param>
    /// <param name="scopes">The scopes to request the access token for.</param>
    public AzureIdentityAuthenticationProvider(TokenCredential credential, string[] allowedHosts, params string[] scopes) : base(new AzureIdentityAccessTokenProvider(credential, allowedHosts, scopes))
    {
    }
}

File: authentication/java/azure/lib/src/main/java/com/microsoft/kiota/authentication/AzureIdentityAuthenticationProvider.java
package com.microsoft.kiota.authentication;

import javax.annotation.Nonnull;


import com.azure.core.credential.TokenCredential;
import com.azure.core.credential.TokenRequestContext;
import com.microsoft.kiota.authentication.BaseBearerTokenAuthenticationProvider;

/** Implementation of AuthenticationProvider that supports implementations of TokenCredential from Azure.Identity. */
public class AzureIdentityAuthenticationProvider extends BaseBearerTokenAuthenticationProvider {
    /**
     * Creates a new instance of AzureIdentityAuthenticationProvider.
     * @param tokenCredential The Azure.Identity.TokenCredential implementation to use.
     * @param allowedHosts The list of allowed hosts for which to request access tokens.
     * @param scopes The scopes to request access tokens for.
     */
    public AzureIdentityAuthenticationProvider(@Nonnull final TokenCredential tokenCredential, @Nonnull final String[] allowedHosts, @Nonnull final String... scopes) {
        super(new AzureIdentityAccessTokenProvider(tokenCredential, allowedHosts, scopes));
    } 
}
